src/datumaro/experimental/fields.py

```diff
@@ -295,8 +308,161 @@ def label_field(
         dtype: Polars data type for label values (defaults to pl.Int32())
         semantic: Semantic tags describing the label purpose (optional)
         multi_label: Whether this field should handle multiple labels (defaults to False)
+        is_list: Whether this field should be treated as a list type (defaults to False)
 
     Returns:
         LabelField instance configured with the given parameters
     """
-    return LabelField(semantic=semantic, dtype=dtype, multi_label=multi_label)
+    return LabelField(semantic=semantic, dtype=dtype, multi_label=multi_label, is_list=is_list)
+
+
+def convert_numpy_object_array_to_series(data):
+    """
+    Convert ragged numpy object arrays to Polars Series recursively.
+
+    Handles nested object arrays containing variable-length lists.
+
+    Example:
+        >>> import numpy as np
+        >>> ragged = np.array([
+        ...     np.array([1, 2, 3]),
+        ...     np.array([4, 5]),
+        ...     np.array([6, 7, 8, 9])
+        ... ], dtype=object)
+        >>> series = convert_numpy_object_array_to_series(ragged)
+        >>> print(series)
+        shape: (3,)
+        Series: '' [list[i64]]
+        [
+                [1, 2, 3]
+                [4, 5]
+                [6, 7, … 9]
+        ]
+
+        # Compare with direct conversion which results
+        # into an object Series instead of a list Series:
+        >>> direct = pl.Series(ragged)
+        >>> print(direct)
+        shape: (3,)
+        Series: '' [o][object]
+        [
+                [1 2 3]
+                [4 5]
+                [6 7 8 9]
+        ]
+    """
+    if data.dtype == "O":
```

Comment:
Please define a variable for this magic string and also reuse it throughout this PR

Reply:
I have replaced it with `object` to avoid the string. It’s equivalent and more explicit.